### ISSUE TITLE
TIFF JPEG compression type resulting in CMYK color space?

### ISSUE THREAD

**Issue by Echoloc8** [Beyka/Android-TiffBitmapFactory#14]
We use **com.sun.imageio.plugins.jpeg.JPEGImageReader** to decode the TIFFs we receive in one portion of our app. I'm using TiffBitmapFactory to generate multipage grayscale TIFFs and using **CompressionScheme.COMPRESSION_JPEG** when saving them. JPEGImageReader is complaining about an unsupported image type, and the one that it's getting from the image is **com.sun.imageio.plugins.jpeg.JPEG.JCS_CMYK**, which it doesn't handle. 

Is there a way to make TiffBitmapFactory emit RGB? Alternately, could TBFactory be outputting CMYK because of anything I'm doing?

**Comment by Beyka** [comment]
Sorry, but I didn't understand your question at all. Can you rephrase your question?
>> is there a way to make TiffBitmapFactory emit RGB? Alternately, could TBFactory be outputting CMYK because of anything I'm doing? 

TiffBitmapFactory is decoder and it always returns android Bitmap class. Every pixel in Bitmap is in ARGB format.
Any way there is no possibility to save CMYK image, because android not suuport CMYK. Every image that you save will be in RGB format and every image that you decode will be in RGB format.

>>I'm using TiffBitmapFactory to generate multipage grayscale TIFFs

grayscale or binary?

**Comment by Echoloc8** [comment]
Okay, this tells me what I worried about, which is that the com.sun.imageio.plugins.jpeg.JPEGImageReader library class we're using is detecting CMYK falsely. I will investigate from that angle. Thank you!

(Also, it's 8-bit grayscale, not binary. When we save with group-4 compression all is well, but we need true grayscale for this project.)

**Comment by Beyka** [comment]
Yeah, sorry if my answer was not very helpful.
If you will have other questions - your welcome.
